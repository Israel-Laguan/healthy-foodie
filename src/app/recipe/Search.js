import React from "react";
import { Row, Col, Button, FormGroup, Input } from "reactstrap";

function Search({ recipe, onChange, onClick }) {
  return (
    <nav className="navbar navbar-dark bg-primary">
      
      <Row form>
        <Col md={11}>
          <FormGroup>
            <Input
              type="text"
              value={recipe}
              placeholder="Buscador recetas"
              onChange={(e) => onChange(e.target.value)}
            />
          </FormGroup>
        </Col>
        <Col md={1}>
          <Button onClick={onClick} className="d-flex ">
            <img
              src="https://img.icons8.com/doodle/48/000000/search--v1.png"
              alt="Search Button"
            />
            Find
          </Button>
        </Col>
      </Row>
    </nav>
  );
}

export default Search;
