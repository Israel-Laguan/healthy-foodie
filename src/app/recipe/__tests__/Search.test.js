import React from 'react';
import renderer from 'react-test-renderer';
import Search from '../Search';

describe('src/app/recipe:details', () => {
  it('render buttons when filters provided', () => {
    const recipe = 'Test Recipe';
    const component = renderer.create(
      <Search recipe={recipe} onChange={a => a} onClick={a => a} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
