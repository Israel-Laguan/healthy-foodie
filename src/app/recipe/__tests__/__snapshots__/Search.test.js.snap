// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/app/recipe:details render buttons when filters provided 1`] = `
<form
  className="form-inline"
>
  <div
    className="d-flex form-group"
  >
    <input
      className="form-control-lg form-control"
      onChange={[Function]}
      placeholder="Search a Recipe"
      type="text"
      value="Test Recipe"
    />
    <button
      aria-label={null}
      className="d-flex p-0 btn btn-secondary"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="Search Button"
        src="https://img.icons8.com/doodle/48/000000/search--v1.png"
      />
    </button>
  </div>
</form>
`;
